oasis/models/model.py

```diff
@@ -2,23 +2,66 @@
 
 from __future__ import absolute_import
 
+import yaml
+from threading import Event, Lock
 from oasis.libs.log import logger
-from oasis.datasource.data_model import DataSource
-# import json
+from oasis.models.util import sub_id
 
 MODELS_PATH = ""
 
 
 class Model(object):
     """Model is a abstract of calculation model."""
     def __init__(self):
+        self.threads = dict()
+        self.event = Event()
+        self.lock = Lock()
+        self._exit = False
         self.model_path = MODELS_PATH
 
     def run(self):
-        logger.info("running calculation model")
+        logger.info("run model")
 
     def close(self):
-        logger.info("close model")
+        with self.lock:
+            self._exit = True
+            self.event.set()
+            for key in self.threads:
+                self.threads[key].join()
 
 
+class Config(object):
+    def __init__(self):
+        self.model = dict()
+        self.metrics = dict()
+
```

Comment:
why need another init

Reply:
abstract the same code and Config can be rewrited.

Reply:
Why not `def __init__(self, config_file, config=None):`?